hangups/conversation.py

```diff
@@ -620,15 +621,12 @@ def _get_event_request_header(self):
                         ),
                         include_event=True,
                         max_events_per_conversation=max_events,
-                        event_continuation_token=(
-                            hangouts_pb2.EventContinuationToken(
-                                event_timestamp=parsers.to_timestamp(
-                                    conv_event.timestamp
-                                )
-                            )
-                        )
+                        event_continuation_token=self._event_cont_token
                     )
                 )
+                self._event_cont_token = (
+                    res.conversation_state.event_continuation_token
+                )
```

Comment:
With this change, each `get_events` call uses the event continuation token saved from the previous one. This happens to be what we want for the UI, but it's inconsistent with the method's interface, which says it returns events preceding the provided event ID.

How about refactoring this method to replace the `event_id` argument with the event continuation token?

Reply:
I've given this some thought, and I wonder if it makes sense to expose the ```EventContinuationToken``` to the user since we don't know how to deterministically generate such a token from an ```event_id```. That is, we could envision a scenario where a user requests events older than an ```event_id``` which isn't the oldest (e.g., requesting events older than event 47 when only the first 50 have been received), but it would not be possible to do the same using the continuation token (in the same example, the token received could only be used to request events 51-100). 

I believe it would be more straightforward to keep the token internal to ```Conversation``` objects, using them only to pull events from the server as needed.

Reply:
I was thinking we'd remove the `event_id` parameter entirely, so the only option would be loading events sequentially using tokens. However, you're right that it's simpler to keep it internal, and there's no immediate requirement to expose it.